kythe/java/com/google/devtools/kythe/extractors/shared/FileVNames.java

```diff
@@ -123,6 +123,11 @@ public BaseFileVName(Pattern pattern, VNameTemplate vname) {
       this.pattern = pattern;
       this.vname = vname;
     }
+
+    // GSON-required no-args constructor.
```

Comment:
Why are these suddenly required?  What is GSON going to do with such a constructor and bunch of `final` fields?

Reply:
I'm not sure what caused these to suddenly break, but technically this was always required by GSON, it just "happened to work" before.  There are fair number of SO questions and GitHub issues around it, so it seemed best to add.